### PR TITLE
MAINT: Gitpod fixes

### PR DESCRIPTION
**Pull request by trallard**
#### What does this implement/fix?

Removing the last few issues in the Gitpod workspace
- Adds comments to the GH actions files
- Adds Git graph VSCode extension (quite nifty for folks getting used to git)
- Ensures ` "restructuredtext.confPath": ""` to use docutils by default
- Removes deprecated settings
- Now ccache is working properly

#### Additional information

Link to my workspace:https://gitpod.io#snapshot/80ff27c9-e202-42b9-b5fc-b7fb06d49dc9

### CODE REVIEW

File: .github/workflows/gitpod.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ name: Build Gitpod Docker image
 on:
   schedule:
     - cron: "0 0 * * *" # nightly
+  push:
+    branches:
+      - feature/**
```
Comment (rgommers) [inline .github/workflows/gitpod.yml:8]: Did you mean to leave this in? Looks like it's for testing on your branch?

File: tools/docker_dev/.gitpod.Dockerfile [hunk 2]
```diff
@@ -1,8 +1,16 @@
+# Doing a local shallow clone - keeps the container secure
+# and much slimmer than usins COPY directly
```
Comment (rgommers) [inline tools/docker_dev/.gitpod.Dockerfile:2]: minor: typo in `usins`

Comment (rgommers) [reply tools/docker_dev/.gitpod.Dockerfile:2]: Out of interest: why is it more secure?

**Review by rgommers — COMMENTED** [review]
Great, thanks Tania. My main question is about the `feature/**` addition, the rest LGTM.

**Review by rgommers — APPROVED** [review]
LGTM now, let's give this a go!

File: tools/docker_dev/.gitpod.Dockerfile (hunk 2 above)
Comment (trallard) [reply tools/docker_dev/.gitpod.Dockerfile:2]: ah I deleted a bit, should be:

`# and much slimmer than using COPY directly or remote clone`

that makes more sense: more secure than the remote clone as you do not pass tokens or have to persist keys with buildkit

Comment (rgommers) [reply tools/docker_dev/.gitpod.Dockerfile:2]: makes sense now, thanks

**Comment by rgommers** [comment]
Still failing, with a different error now (https://github.com/scipy/scipy/runs/2395905565):
```
 > [build 4/5] RUN conda activate scipy-dev &&     python setup.py build_ext --inplace &&     ccache -s:
------
.gitpod.Dockerfile:38
--------------------
  37 |     # Must re-activate conda to ensure the ccache flags are picked up
  38 | >>> RUN conda activate ${CONDA_ENV} && \
  39 | >>>     python setup.py build_ext --inplace && \
  40 | >>>     ccache -s
  41 |     
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/bash --login -o pipefail -c conda activate ${CONDA_ENV} &&     python setup.py build_ext --inplace &&     ccache -s]: exit code: 1
```

**Comment by trallard** [comment]
This is weird it comes up with `Could not find conda environment: scipy-dev`
 but it was definitely working on my fork. Will have a quick look - for reference a successful run: https://github.com/trallard/scipy/actions/runs/764695150

**Comment by trallard** [comment]
Found it: The `scipy/scipy-dev:latest` in DockeHub returns this issue:

```
(base) root@8abf2acdb709:/home/scipy# conda activate scipy-dev
Could not find conda environment: scipy-dev
You can list all discoverable environments with `conda info --envs`.

(base) root@8abf2acdb709:/home/scipy#
```

Might have been overwritten at some point will push an updated image to DockerHub now

**Comment by rgommers** [comment]
I'm still getting confused, so once it works I may try to write some developer docs to explain it. Just the very basics for myself now:

- `.github/workflows/docker.yml` builds `scipy/scipy-dev` images (both `20210417-master-04ecc30f` ones, and `latest`). These contain the development environment (all `apt` and `conda` installs, and the `scipy-dev` conda env populated).
- `.github/workflows/gitpod.yml` builds `scipy/scipy-gitpod` images. Should be `latest` tag and date tags as well. This Docker Hub repository doesn't exist yet, because no build has succeeded.
- `.gitpod.yml` then simply pulls `image: scipy/scipy-gitpod:latest` to get started inside a Gitpod workspace.

**Comment by trallard** [comment]
Attempting to clarify the workflow here: 


Gitpod

**Comment by trallard** [comment]
Now that I think about it I said I'd remove the push on `master` for `scipy/scipy-dev:latest`

**Comment by rgommers** [comment]
Very nice, that's super clear!

**Pull request merged**